File: Kernel Simulator/Network/RemoteDebug/Interface/RemoteDebugCommandExecutor.vb

'    Kernel Simulator  Copyright (C) 2018-2022  Aptivi
'
'    This file is part of Kernel Simulator
'
'    Kernel Simulator is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Kernel Simulator is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Imports System.IO

Namespace Network.RemoteDebug.Interface
    Public MustInherit Class RemoteDebugCommandExecutor
        Inherits CommandExecutor
        Implements IRemoteDebugCommand

        Public Overridable Overloads Sub Execute(StringArgs As String, ListArgsOnly() As String, ListSwitchesOnly() As String, SocketStreamWriter As StreamWriter, DeviceAddress As String) Implements IRemoteDebugCommand.Execute
            Wdbg(DebugLevel.F, "We shouldn't be here!!!")
            Throw New InvalidOperationException()
        End Sub

    End Class
End Namespace

File: public/Kernel Simulator/Kernel/Debugging/RemoteDebug/Interface/RemoteDebugCommandExecutor.cs

// Kernel Simulator  Copyright (C) 2018-2022  Aptivi
// 
// This file is part of Kernel Simulator
// 
// Kernel Simulator is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// Kernel Simulator is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.

using System;
using System.IO;
using KS.Shell.ShellBase.Commands;

namespace KS.Kernel.Debugging.RemoteDebug.Interface
{
    /// <summary>
    /// Remote debug command executor
    /// </summary>
    public abstract class RemoteDebugCommandExecutor : CommandExecutor, IRemoteDebugCommand
    {

        /// <inheritdoc/>
        public virtual void Execute(string StringArgs, string[] ListArgsOnly, string[] ListSwitchesOnly, StreamWriter SocketStreamWriter, string DeviceAddress)
        {
            DebugWriter.WriteDebug(DebugLevel.F, "We shouldn't be here!!!");
            throw new InvalidOperationException();
        }

    }
}
